### ISSUE TITLE
Start implementing BIDS URIs

### ISSUE THREAD

**Issue by hoechenberger** [mne-tools/mne-bids#860]
There's a [draft](https://github.com/bids-standard/bids-specification/pull/820/files) for a huge BIDS specification update that will add URIs. I'm feeling that we should start implementing this before the draft gets final approval, so we can see where potential issues both for devs and users might occur. I'm worried we'll get this fancy BIDS enhancement only to discover that is has some serious flaws we simply overlooked because we never tried putting it to use in real-world scenarios.

**Comment by sappelhoff** [comment]
Agreed, thanks for opening this issue @hoechenberger

**Comment by adam2392** [comment]
How should we start implementing this? Some ideas that have popped into my head over the months have been:

#### For writing derivatives
- overload `write_raw_bids` even further, adding some kwargs to allow writing derived datasets that are "sourced" from another
- start developing a whole new API dedicated to derivatives, such as `make_derivative_dataset`, which will define sources/rawsources, create the dataset_description.json internally, 

### For reading derivative stuff
In order to really make derived datasets useful in mne/mne-bids context, I think perhaps even using the `mne.Info` object as a way to pass "raw EEG source metadata" around. E.g. if you have a derived dataset that needs access to a consolidated data structure containing the entire source metadata, then you can easily load up an `Info` object?

**Comment by sappelhoff** [comment]
note: https://github.com/bids-standard/bids-specification/pull/918 has been merged and BIDS URIs will be released sometime in september with BIDS 1.8 (which will contain fNIRS).

The final proposal for BIDS URIs was very simple and straight forward (IMHO) and it should be quite easy to implement in mne-bids: https://bids-specification.readthedocs.io/en/latest/02-common-principles.html#bids-uri

**Comment by sappelhoff** [comment]
yes, we have accumulated a backlog on issues that are actually important to work on. Like this one, and migrating to the new validator (#1288)
